src/backend/catalog/pg_compression.c

```diff
@@ -437,10 +554,11 @@ compresstype_is_valid(char *comptype)
 					 cql("SELECT COUNT(*) FROM pg_compression "
 						 " WHERE compname = :1 ",
 						 NameGetDatum(&compname))));
-		
+
+	/* FIXME: This is a hack. Should register gzip handlers into pg_compression table. */
 	if(!found)
 	{
-		if((strcmp(comptype, "snappy") == 0) || strcmp(comptype, "gzip") == 0)
+		if(strcmp(comptype, "gzip") == 0)
```

Comment:
It's better we remove this hack finally.  Since this commit is for snappy support for AO, let's move removing gzip hack to another JIRA.